packages/models/files.js

```diff
@@ -7,6 +7,12 @@ const Files = new FilesCollection({
     debug: true,
     collectionName: 'files',
     protected(fileObj) {
+        var session = this.params.query.xmtok;
+        var userId  = (Meteor.server.sessions[session] && Meteor.server.sessions[session].userId) ? Meteor.server.sessions[session].userId : null;
```

Comment:
Would this work on multiple nodes within one scaleset? `Meteor.server.sessions[session]` looks like it's a local object, it might be better to fetch the token from the database instead.